presto-spi/src/main/java/com/facebook/presto/spi/type/TypeSignature.java

```diff
@@ -35,6 +37,12 @@
     private final List<TypeSignatureParameter> parameters;
     private final boolean calculated;
 
+    private static final Map<String, String> baseNameAliasToCanonical = new TreeMap<String, String>();
```

Comment:
here one could use `TreeMap` with case insensitive comparator (`new TreeMap(String.CASE_INSEN...)`). Otherwise the fact keys are lowercased is in two places: map contents and where lookup is actually done.